src/main/java/nl/opengeogroep/safetymaps/server/stripes/SafetyConnectProxyActionBean.java

```diff
@@ -70,56 +83,143 @@ public Resolution proxy() throws Exception {
-                    String encoding = "UTF-8";
-                    response.setCharacterEncoding(encoding);
-                    response.setContentType(contentType.getValue());
-
-                    OutputStream out;
-                    String acceptEncoding = request.getHeader("Accept-Encoding");
-                    if(acceptEncoding != null && acceptEncoding.contains("gzip")) {
-                        response.setHeader("Content-Encoding", "gzip");
-                        out = new GZIPOutputStream(response.getOutputStream(), true);
-                    } else {
-                        out = response.getOutputStream();
+
+                return new Resolution() {
+                    @Override
+                    public void execute(HttpServletRequest request, HttpServletResponse response) throws Exception {
+                        String encoding = "UTF-8";
+
+                        response.setCharacterEncoding(encoding);
+                        response.setContentType(contentType.getValue());
+
+                        OutputStream out;
+                        String acceptEncoding = request.getHeader("Accept-Encoding");
+                        if(acceptEncoding != null && acceptEncoding.contains("gzip")) {
+                            response.setHeader("Content-Encoding", "gzip");
+                            out = new GZIPOutputStream(response.getOutputStream(), true);
+                        } else {
+                            out = response.getOutputStream();
+                        }
+                        IOUtils.copy(new StringReader(authorizedContent), out, encoding);
+                        out.flush();
+                        out.close();
                     }
-                    IOUtils.copy(new StringReader(content), out, encoding);
-                    out.flush();
-                    out.close();
+                };
+            } catch(IOException e) {
+                log.error("Failed to write output:", e);
+                return null;
+            }
+
+        } catch(Exception e) {
+            return new StreamingResolution("application/json", logExceptionAndReturnJSONObject(log, "Error on " + url + "/" + path, e).toString());
+        }
+    }
+
+    private String applyAuthorizationToIncidentContent(String contentFromResponse, Connection c) throws Exception {
+        HttpServletRequest request = context.getRequest();
+
+        boolean kladbklokAlwaysAuthorized = "true".equals(Cfg.getSetting("kladblok_always_authorized", "false"));
+        boolean incidentMonitorKladblokAuthorized = kladbklokAlwaysAuthorized || request.isUserInRole(ROLE_ADMIN) || request.isUserInRole(ROLE_INCIDENTMONITOR_KLADBLOK);
+        boolean eigenVoertuignummerAuthorized = request.isUserInRole(ROLE_ADMIN) || request.isUserInRole(ROLE_EIGEN_VOERTUIGNUMMER);
+        boolean incidentMonitorAuthorized = request.isUserInRole(ROLE_ADMIN) || request.isUserInRole(ROLE_INCIDENTMONITOR);
```

Comment:
Early return possible when `incidentMonitorAuthorized` is true, makes the rest of the filtering simpler and does not require the `getUserDetails()` DB query.